homeassistant/components/shopping_list/__init__.py

```diff
@@ -351,7 +350,7 @@ def async_reorder(
                 )
             new_items.append(value)
         self.items = new_items
-        self.hass.async_add_executor_job(self.save)
+        await self.hass.async_add_executor_job(self.save)
```

Comment:
I don't think there's any reason to not await here and in the sort case considering we do it in all other cases.